File: Angular/src/app/models/Message.ts
export class Message {
    id: number;
    chatId: number;
    userId: number;
    previousMessageId: number;
    content: string;
    selfContent: string | null;
    urlLink: string;
    time: Date;
    isEdited: boolean;
    isPinned: boolean;
}

export class MessageDTO {
  userId: number;
  chatId: number;
  content: string;
  replyMessageId: number | null;
}

File: WebApplication1/WebApplication1/Models/Data/Message.cs
namespace WebApplication1.Models.Data
{
    public class Message
    {
        public int Id { get; set; } 

        public int ChatId { get; set; }

        public int UserId { get; set; }

        public int? PreviousMessageId { get; set; }

        public string Content { get; set; }

        public string? SelfContent { get; set; }

        public string? URLLink { get; set; }

        public DateTime Time {  get; set; }

        public bool IsEdited { get; set; }

        public bool IsPinned { get; set; }
    }
}
